File: php/basic-example.php
<?php
/**
 * GenelPara API - Basit Kullanım Örneği
 * 
 * Bu örnek, GenelPara API'sini kullanarak döviz ve kripto para
 * fiyatlarını çekmeyi gösterir.
 */

// API endpoint
$apiUrl = 'https://api.genelpara.com/json/';

// İstek parametreleri
$params = http_build_query([
    'list' => 'doviz',
    'sembol' => 'USD,EUR,GBP'
]);

// cURL ile istek gönder
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $apiUrl . '?' . $params);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_TIMEOUT, 10);

$response = curl_exec($ch);
$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

if (curl_errno($ch)) {
    die('cURL Hatası: ' . curl_error($ch));
}

curl_close($ch);

// Yanıtı işle
if ($httpCode == 200) {
    $data = json_decode($response, true);
    
    if ($data['success']) {
        echo "=== DÖVİZ FİYATLARI ===\n\n";
        
        foreach ($data['data'] as $symbol => $info) {
            echo sprintf(
                "%s: Alış: %s %s | Satış: %s %s | Değişim: %s%%\n",
                $symbol,
                $info['alis'],
                $info['sembol'],
                $info['satis'],
                $info['sembol'],
                $info['degisim']
            );
        }
        
        echo "\n=== RATE LIMIT BİLGİSİ ===\n";
        echo sprintf(
            "Kalan İstek: %d/%d\n",
            $data['rate_limit']['remaining'],
            $data['rate_limit']['limit']
        );
        echo "Sıfırlanma Zamanı: " . $data['rate_limit']['reset_at'] . "\n";
    } else {
        echo "API Hatası: " . ($data['error'] ?? 'Bilinmeyen hata') . "\n";
    }
} else {
    echo "HTTP Hatası: " . $httpCode . "\n";
}

File: python/basic_example.py
#!/usr/bin/env python3
"""
GenelPara API - Python Basit Kullanım Örneği

Bu örnek, GenelPara API'sini kullanarak döviz ve kripto para
fiyatlarını çekmeyi gösterir.
"""

import requests
import json
from typing import Optional, Dict, Any

API_URL = 'https://api.genelpara.com/json/'

def get_rates(list_type: str, symbols: str) -> Optional[Dict[Any, Any]]:
    """
    API'den veri çek
    
    Args:
        list_type: Kategori (doviz, kripto, altin, vb.)
        symbols: Sembol listesi (virgülle ayrılmış) veya 'all'
    
    Returns:
        API yanıtı veya None
    """
    try:
        params = {
            'list': list_type,
            'sembol': symbols
        }
        
        response = requests.get(API_URL, params=params, timeout=10)
        response.raise_for_status()
        
        return response.json()
    
    except requests.exceptions.RequestException as e:
        print(f"❌ API isteği başarısız: {e}")
        return None
    except json.JSONDecodeError as e:
        print(f"❌ JSON parse hatası: {e}")
        return None

def display_rates(data: Dict[Any, Any]) -> None:
    """Verileri formatlı olarak göster"""
    
    if not data or not data.get('success'):
        print(f"❌ API hatası: {data.get('error', 'Bilinmeyen hata')}")
        return
    
    print("\n=== DÖVİZ KURLARI ===\n")
    
    for symbol, info in data['data'].items():
        arrow = '↗' if info['yon'] == 'moneyUp' else '↘' if info['yon'] == 'moneyDown' else '→'
        
        print(f"{symbol:6s}: {float(info['satis']):>10.4f} {info['sembol']} "
              f"{arrow} {info['degisim']:>8s} ({info['oran']:>6s}%)")
    
    print("\n=== RATE LIMIT ===")
    rate_limit = data['rate_limit']
    print(f"Kalan: {rate_limit['remaining']}/{rate_limit['limit']}")
    print(f"Sıfırlanma: {rate_limit['reset_at']}")

def main():
    """Ana fonksiyon"""
    print("GenelPara API'den veri çekiliyor...\n")
    
    # Döviz kurlarını çek
    data = get_rates('doviz', 'USD,EUR,GBP,JPY')
    
    if data:
        display_rates(data)

if __name__ == '__main__':
    main()
